src/main/scala/chiseltest/coverage/midas/Builder.scala

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiseltest.coverage.midas
+
+import firrtl._
+
+/** Helps us construct well typed low-ish firrtl.
+  * Some of these convenience functions could be moved to firrtl at some point.
+  */
+object Builder {
+  def findClock(m: ir.Module): ir.Expression = {
```

Comment:
I'm not totally following this, but this would also be vulnerable when registers whose clocks wires have been casted with `.asClock`

Reply:
Oh yeah, there is no multi-clock support yet. I will add that to the known limitations.